Compilation failure with GCC 15 + cmake - bool is now a keyword (C23+)

When attempting to compile OpenSimH with GNU C++ 15.2.1 (Fedora 42) using the cmake script, I got the following failure:

```text
In file included from /simh/slirp/slirp.h:4,
                 from /simh/slirp/arp_table.c:25:
/simh/slirp_glue/config-host.h:15:13: error: ‘bool’ cannot be defined via ‘typedef’
   15 | typedef int bool;
      |             ^~~~
/simh/slirp_glue/config-host.h:15:13: note: ‘bool’ is a keyword with ‘-std=c23’ onwards
/simh/slirp_glue/config-host.h:15:1: warning: useless type name in empty declaration
   15 | typedef int bool;
      | ^~~~~~~
gmake[2]: *** [slirp/CMakeFiles/slirp.dir/build.make:79: slirp/CMakeFiles/slirp.dir/arp_table.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2645: slirp/CMakeFiles/slirp.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2
*** cmake/cmake-builder.sh: Build errors detected. Exiting.
```

We can either dig into the source and fix it in a portable way, or ask the GNU compiler to check against a less strict standard.

--- a/.travis/deps.sh
+++ b/.travis/deps.sh
@@ -17,6 +17,7 @@ install_arch_linux() {
     sudo pacman -S --noconfirm mesa
     sudo pacman -S --noconfirm libsm
     sudo pacman -S --noconfirm cmake
+    sudo pacman -S --noconfirm sdl2_ttf
 
 }
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,9 @@ endif()
 
 project(simh VERSION "${SIMH_VERSION}" LANGUAGES C CXX)
 
+## Tell cmake we need c99 because GCC 15 defaults to c23 which breaks things.
+set(CMAKE_C_STANDARD 99)
+
 include(vcpkg-setup)
 include(GNUInstallDirs)
 

